### PR TITLE
Adding Examples of single event runs

### PR DESCRIPTION
**Pull request by xuyuon**
- Fixed minor bugs in the test script which could yield `nan` in Adam optimization and ruin the run.
- Added examples of single event runs: `GW150914_D.py`, `GW150914_D_heterodyne.py`, `GW150914_Pv2.py`

### CODE REVIEW

File: src/jimgw/transforms.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def transform(self, x: dict[str, Float]) -> tuple[dict[str, Float], Float]:
         output_params = self.transform_func(transform_params)
         jacobian = jax.jacfwd(self.transform_func)(transform_params)
         jacobian = jnp.array(jax.tree.leaves(jacobian))
-        jacobian = jnp.log(jnp.linalg.det(jacobian.reshape(self.n_dim, self.n_dim)))
+        jacobian = jnp.log(jnp.absolute(jnp.linalg.det(jacobian.reshape(self.n_dim, self.n_dim))))
```
[the remark this answers is not in the document]
Comment (xuyuon) [reply src/jimgw/transforms.py:92]: yes sure

[the remark this answers is not in the document]
Comment (xuyuon) [reply src/jimgw/transforms.py]: It should be good to go now

**Comment by thomasckng** [comment]
Btw, we may want to remove legacy examples either in this PR or another new PR.

File: example/GW150914_D.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    BoundToUnbound(name_mapping = [["q"], ["q_unbounded"]], original_lower_bound=q_min, original_upper_bound=q_max),
+    BoundToUnbound(name_mapping = [["s1_z"], ["s1_z_unbounded"]] , original_lower_bound=-1.0, original_upper_bound=1.0),
+    BoundToUnbound(name_mapping = [["s2_z"], ["s2_z_unbounded"]] , original_lower_bound=-1.0, original_upper_bound=1.0),
+    BoundToUnbound(name_mapping = [["d_L"], ["d_L_unbounded"]] , original_lower_bound=1.0, original_upper_bound=2000.0),
+    BoundToUnbound(name_mapping = [["t_c"], ["t_c_unbounded"]] , original_lower_bound=-0.05, original_upper_bound=0.05),
+    BoundToUnbound(name_mapping = [["phase_c"], ["phase_c_unbounded"]] , original_lower_bound=0.0, original_upper_bound=2 * jnp.pi),
+    BoundToUnbound(name_mapping = [["iota"], ["iota_unbounded"]], original_lower_bound=0., original_upper_bound=jnp.pi),
+    BoundToUnbound(name_mapping = [["psi"], ["psi_unbounded"]], original_lower_bound=0.0, original_upper_bound=jnp.pi),
+    SkyFrameToDetectorFrameSkyPositionTransform(name_mapping = [["ra", "dec"], ["zenith", "azimuth"]], gps_time=gps, ifos=ifos),
```
Comment (kazewong) [inline example/GW150914_D.py:75]: This does not work with the new predefined transform. Please change accordingly

**Comment by xuyuon** [comment]
I am going to close this PR, and separate it into two new PR.

File: example/GW150914_D.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+fmax = 1024.0
+
+ifos = [H1, L1]
+
+for ifo in ifos:
+    ifo.load_data(gps, start_pad, end_pad, fmin, fmax, psd_pad=16, tukey_alpha=0.2)
+
+M_c_min, M_c_max = 10.0, 80.0
+q_min, q_max = 0.125, 1.0
+m_1_prior = UniformPrior(Mc_q_to_m1_m2(M_c_min, q_max)[0], Mc_q_to_m1_m2(M_c_max, q_min)[0], parameter_names=["m_1"])
```
[the remark this answers is not in the document]
Comment (xuyuon) [reply example/GW150914_D.py:36]: I think this is related to the fact that when you sample from `UniformPrior`, it just randomly gives you a value within the range you have defined. Now, without any constraint, the range of `m_1` and `m_2` overlaps with each other, so `m_2` can be bigger than `m_1` when you try to sample on it. When `m_2` is bigger than `m_1`, it gives strange value of `M_c` and `q`.

I think its better to build a special prior class for `m_1` and `m_2`, in which we add the constraint `m_1` > `m_2` in sampling.

[the remark this answers is not in the document]
Comment (thomasckng) [reply example/GW150914_D.py:36]: I think the value looks fine to me. I can test it with @xuyuon tomorrow.

**Comment by xuyuon** [comment]
This PR is replaced by #157 and #154

**Pull request closed without merging**